libmscore/rendermidi.cpp

```diff
@@ -701,6 +701,48 @@ static void collectMeasureEventsDefault(EventMap* events, Measure* m, Staff* sta
             Fraction tick = seg->tick();
             Fraction tick2 = tick + seg->ticks();
 
+            //render harmony
```

Comment:
Looks pretty much exactly like I imagined, does it work well?  I haven't gotten around to building this branch but plan to soon.

I guess it might be worth factoring this out to a separate function.  Maybe it would also need to be called from collectMeasureEventsSimple()?  That split is recent, and I'm not at all familar with how the different methods of handling dynamics interact.  @jthistle ?

Reply:
It works, but I don't think it will work for all cases yet. I know transposition won't work yet, but that should be fairly easy to account for. I'll make these adjustments and factor out the code to a separate function. Also as it is right now the chord symbols are played using the same part that the chord symbol is attached to. It's not too big of a deal, but it means that there isn't yet a way to play the harmony with a different instrument than the part. I'm looking into that though.

Reply:
Also there's a bit of a bigger issue where the annotations aren't registered as rendered/unrendered (which they're annotations so yeah that's normal) so if the score is full of rests chord symbols won't play. I've looked into it, but I couldn't find a way to quickly fix it so I just continued implementing it as is for now. I'll figure that out sometime also.

Reply:
There are lots of examples where we have a few lines of code you can steal to handle the transposition (check for concert pitch state, grab transposition interval, transpose by that).  As for playback sound, I guess I can see why you might want piano or guitar playback even though chords are attached to some other instrument, but I wouldn't worry about that.  Actually, lead sheets are alrady piano by default so those are good.  Maybe you also need to make sure only one chord symbol per segment gets played, though - there might be several staves that include them (solo parts plus guitar, piano, and bass in a big band chart, for instance).  How to decide which? If nothing else, there should be a "Play" checkbox in Inspector you can uncheck.  But maybe there should be other hueristics as well.

Reply:
I am not understanding the bit about annotations being registered or why rests vs notes matters here.

Reply:
Oh the rendering system only goes through what it thinks needs to be rendered so if there are no notes in the score or chunk, then it doesn't render it and so the chord symbols also are not rendered. I'm not completely sure how these things are determined yet though so I will have to look into it more. It's not really a showstopper right now since the problem case is essentially if you have no notes on the score at all. Otherwise one note will allow all chord symbols to be played as far as I'm aware. It's just something I noticed and I'll probably get to it after I make a little more progress with midi rendering.